Renames RigidBodyTreeLcmPublisher to be DrakeVisualizer.

liangfok opened the pull request:
Currently, the messages published by this system are only used by `drake-visualizer`. Thus, it makes sense for the system's name to include the word "Visualizer".

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/robotlocomotion/drake/3954)
<!-- Reviewable:end -->

liangfok commented:
+@russtedrake for feature review, please.
+@sherm1 for platform review, please.

/CC: @amcastro-tri 

---

Review status: 0 of 12 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-:-KVIiuwHgsKzwBrMVJLO:b1wea1o)*
<!-- Sent from Reviewable.io -->

RussTedrake commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 15 of 15 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/automotive_simulator.cc, line 194 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-KVKr44V3ARd9eiTdutC:-KVKr44WlRcMDr2ORGzO:bwy9efh) ([raw file](https://github.com/robotlocomotion/drake/blob/64eaa81873f0319dceadc92f306147dab2ead91c/drake/automotive/automotive_simulator.cc#L194)):*
> ```cpp
>   DRAKE_DEMAND(!started_);
> 
>   if (!rigid_body_tree_publisher_inputs_.empty()) {
> ```

:OK: note: old name appears here, too.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-:-KVKrRi6YR1BvlqJ4iRl:bnfp4nl)*
<!-- Sent from Reviewable.io -->

liangfok commented:
@russtedrake I addressed your comment. PTAL.

---

Review status: 11 of 13 files reviewed at latest revision, 1 unresolved discussion.

---

*[drake/automotive/automotive_simulator.cc, line 194 at r1](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-KVKr44V3ARd9eiTdutC:-KVLmFkMjEXznP6bH9s-:b-896fix) ([raw file](https://github.com/robotlocomotion/drake/blob/64eaa81873f0319dceadc92f306147dab2ead91c/drake/automotive/automotive_simulator.cc#L194)):*
<details><summary><i>Previously, RussTedrake (Russ Tedrake) wrote…</i></summary>
> :OK: note: old name appears here, too.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-:-KVLmVuLoTlT2oVdRRy6:b-gzzgdn)*
<!-- Sent from Reviewable.io -->

RussTedrake commented:
my comment was marked :OK:, no? still <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-:-KVM2vC5muHRQ6bDBO_e:b5cwqj9)*
<!-- Sent from Reviewable.io -->

liangfok commented:
Sorry, I now see that it was marked :OK:. Actually, I believe the correct markup is `:ok:` (lowercase).

---

Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-:-KVMJfscFPLrS39Q-OGt:b-3bky4g)*
<!-- Sent from Reviewable.io -->

sherm1 commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 14 of 15 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/robotlocomotion/drake/3954#-:-KVMdKFWmY7Ml9Cdk-28:bnfp4nl)*
<!-- Sent from Reviewable.io -->

amcastro-tri commented:
Late to the party. I am happy with this change anyway, I think @RussTedrake has a good argument in thinking this is very drake visualizer specific. Thank you @liangfok for making the change even on a Sunday!
